### DISCUSSION TITLE
Why the changes made are not taking effect in nextjs?

### DISCUSSION THREAD

**Discussion by nhnasem** [vercel/next.js#41238] [Help] [answered]
I installed the latest version of nodejs and using vscode. I'm on a new pc. When I was working with nextjs before, and whenever I made any changes in the code, the localhost representation was instantaneous with a one or two seconds delay.

But now, no matter the changes I make, it doesn't affect the localhost representation. I created a project using 'npx create-next-app new-app' and started the local server 'npm run dev'. 

I also tried deleting the .next folder and rebuilding the project 'npm run build'. Then starting the server again. Still, the changes made in the code doesn't take effect.

What am I doing wrong?

**Comment by kutsoji** [comment, marked as answer]
_make sure you save the file in order to track the changes_

**Reply by nhnasem** [reply]
THanks

**Reply by kutsoji** [reply]
no problem. also don't forget to mark as answered if everything is fine now.

**Comment by icyJoseph** [comment]
Hi,

If you are building `next build` and doing `next start`, it'll run the production, minified, compressed and unchanging version of your application. 

You need to run `next dev` which activates among other things, Hot Module Reloading, which helps you see your changes as you save.

**Comment by Ashish-3000** [comment]
Hi,
I have saved the changes in my vs code but it's still not working in my case.
The same problem occurred to me while I was using Vite but using chaukidar worked in that case.
In the next js --watch is also not working.
